Update changelog for 1.5.0 (#1714)
astronomer/astro-sdk: python-sdk/docs/CHANGELOG.md

# Changelog

## 1.5.0

### Feature:

> - Support Microsoft SQL as Database [#1538](https://github.com/astronomer/astro-sdk/pull/1538)

Author: https://github.com/astronomer/astro-sdk/pull/1714/commits/4d952ac8ca303bc8c2eb7fbfe76a1632e40943d1

> - Support DuckDB as Database. `aql.merge` requires a fix on DuckDB as `UPDATE/IGNORE` clause is not supported as per [bug on duckdb](https://github.com/duckdb/duckdb/issues/6087) [#1695](https://github.com/astronomer/astro-sdk/pull/1695)

Reviewer: ```suggestion
- Support DuckDB as Database. [bug on duckdb](https://github.com/duckdb/duckdb/issues/6087) [#1695](https://github.com/astronomer/astro-sdk/pull/1695)
```

Author: https://github.com/astronomer/astro-sdk/pull/1714/commits/4d952ac8ca303bc8c2eb7fbfe76a1632e40943d1

> - Support DuckDB as Database. `aql.merge` requires a fix on DuckDB as `UPDATE/IGNORE` clause is not supported as per [bug on duckdb](https://github.com/duckdb/duckdb/issues/6087) [#1695](https://github.com/astronomer/astro-sdk/pull/1695)

Reviewer 2: ```suggestion
- Add support for  DuckDB. `aql.merge` requires a fix on DuckDB as `UPDATE/IGNORE` clause is not supported as per [bug on duckdb](https://github.com/duckdb/duckdb/issues/6087) [#1695](https://github.com/astronomer/astro-sdk/pull/1695)
```

Author: https://github.com/astronomer/astro-sdk/pull/1714/commits/4d952ac8ca303bc8c2eb7fbfe76a1632e40943d1

> - Add `load_options` param to accept `pandasOptions` in `load_file` operator [#1466](https://github.com/astronomer/astro-sdk/pull/1466)

Reviewer 3: why we are including this in 1.5.0? We have already released this in 1.4.0 https://github.com/astronomer/astro-sdk/blob/main/python-sdk/docs/CHANGELOG.md#140 cc: @phanikumv

Author: Its part of 1.5.0a1 https://github.com/astronomer/astro-sdk/pull/1666

Author: [Removed](https://github.com/astronomer/astro-sdk/pull/1714/commits/14ec0a8ce30619c465cc4a3981e734b542b65db4) it as this was released and this issue uses this feature: https://github.com/astronomer/astro-sdk/issues/1264

> - Add `load_options` param to accept `pandasOptions` in `load_file` operator [#1466](https://github.com/astronomer/astro-sdk/pull/1466)

Reviewer: @ankit I think this was already done in prior-releases, can we check?

Author: It's part of 1.5.0a1 https://github.com/astronomer/astro-sdk/pull/1666. Isn't it?

Author: [Removed](https://github.com/astronomer/astro-sdk/pull/1714/commits/14ec0a8ce30619c465cc4a3981e734b542b65db4) it as this was released and this issue uses this feature: https://github.com/astronomer/astro-sdk/issues/1264

- Add `result_format` and `fail_on_empty` params to `run_raw_sql` operator [#1584](https://github.com/astronomer/astro-sdk/pull/1584)

> - Add support `validation_mode` as part of the `COPY INTO` command for snowflake. Specify the supported `validation_mode`; `RETURN_n_ROWS` or `RETURN_ERRORS` or `RETURN_ALL_ERRORS`. This instructs the `COPY INTO` command to validate the data files instead of loading them into the specified table; i.e. the `COPY INTO` command tests the files for errors but does not load them. [#1689](https://github.com/astronomer/astro-sdk/pull/1689)

Reviewer: `Add support `validation_mode` as part of the `COPY INTO` command for the snowflake.` i think this should be sufficient.

Author: Should we not make it clear to our users what this change means? Like we did in past like https://github.com/astronomer/astro-sdk/blob/main/python-sdk/docs/CHANGELOG.md#features

Reviewer: IMO, If someone is interested they can follow up on the ticket/PR.

Author: https://github.com/astronomer/astro-sdk/pull/1714/commits/4eb93d4e1542e69c9f763afefd5b62828860ed95

- Add support for native transfers for Azure Blob Storage to Snowflake in `LoadFileOperator`. [#1675](https://github.com/astronomer/astro-sdk/pull/1675)

### Improvements
- Use cache to reduce redundant database calls [#1488](https://github.com/astronomer/astro-sdk/pull/1488)

> - Remove default `copy_options` as part of `SnowflakeLoadOptions`. All `copy_options` are now supported as part of `SnowflakeLoadOptions` as per [documentation](https://docs.snowflake.com/en/sql-reference/sql/copy-into-table.html#copy-options-copyoptions). [#1689](https://github.com/astronomer/astro-sdk/pull/1689)

Reviewer: Do we need to mention this twice?

Author: This is an improvement on `copy_option`. In above we had added a feature for `validation_mode`.

Reviewer: Cool. Make sense.

### Bug fixes

> - When multiple dataframes are passed to `run_raw_sql`, every dataframe should be loaded to different temporary tables. [#1684](https://github.com/astronomer/astro-sdk/pull/1684)

Reviewer: ```suggestion
- Fix handling of multiple dataframes in the `run_raw_sql` operator. [#1684](https://github.com/astronomer/astro-sdk/pull/1684)
```

Author: https://github.com/astronomer/astro-sdk/pull/1714/commits/4d952ac8ca303bc8c2eb7fbfe76a1632e40943d1

# Changelog

## 1.5.0

### Feature:
- Add support for Microsoft SQL server. [#1538](https://github.com/astronomer/astro-sdk/pull/1538)

> - Add support DuckDB. [#1695](https://github.com/astronomer/astro-sdk/pull/1695)

Reviewer: ```suggestion
- Add support for DuckDB. [#1695](https://github.com/astronomer/astro-sdk/pull/1695)
```

Author: https://github.com/astronomer/astro-sdk/pull/1714/commits/16e9d0fbcc8ef7cbc4d39384ef1c30fcaa6c7ce9
